ODCRequest/js/RequestForm.js

```diff
@@ -55,17 +55,17 @@ define(["dojo/_base/declare",
         var _didIteratorError = false;
         var _iteratorError = undefined;
 
-        date_objects = []
+        this.date_objects.length = 0
```

Comment:
I don't think this is needed after you initialise this to an empty array in the line 43

Reply:
If this wasn't here then each time a separate product is clicked the date_objects array will get bigger and bigger. This just ensures that there's an empty array each time